src/handlers/handleCurrentPage/handleCurrentPage.tsx

```diff
@@ -0,0 +1,37 @@
+interface CurrentPageHandler {
+  event: React.MouseEvent<HTMLAnchorElement>;
+  currentPage: number;
+  pageItemsCount: number;
+  setCurrentPage: (pageNumber: number) => void;
+  newParams: URLSearchParams;
+  setSearchParams: (currentOptions: URLSearchParams) => void;
+}
+
+export default function handleCurrentPage(params: CurrentPageHandler): void {
+  const {
+    event,
+    currentPage,
+    pageItemsCount,
+    setCurrentPage,
+    newParams,
+    setSearchParams,
+  } = params;
+  const value = (event.target as HTMLAnchorElement).text;
+
+  if (value === '«' && currentPage > 1) {
+    setCurrentPage(currentPage - 1);
+    newParams.set('page', `${currentPage - 1}`);
+  }
+
+  if (value === '»' && currentPage < pageItemsCount) {
+    setCurrentPage(currentPage + 1);
+    newParams.set('page', `${currentPage + 1}`);
+  }
+
+  if (Number.isInteger(+value)) {
```

Comment:
The condition `Number.isInteger(+value)` is used to check if the `value` is a number. Ensure that `value` is always a valid string representation of a number to avoid unexpected behavior.